rows.go

```diff
@@ -386,5 +389,15 @@ func downloadChunkHelper(ctx context.Context, scd *snowflakeChunkDownloader, idx
 				Message:     errMsgFailedToGetChunk,
 				MessageArgs: []interface{}{idx},
 			}}
+		scd.DoneDownloadCond.Broadcast()
+		scd.ChunksMutex.Unlock()
 	}
 }
+
+func raiseDownloadError(scd *snowflakeChunkDownloader, idx int, err error) {
```

Comment:
Does this actually raise an error? Looks like we're just storing it.

Reply:
yes, storing the error to an error channel and broadcasting to tell the main thread that this chunk loader failed.